src/components/Tokens/TokenRowWallets.tsx

```diff
@@ -217,13 +217,21 @@ export const TokenRowWallets = component$<TokenRowWalletsProps>(
         .value;
       latestTokenPrice.value = (await data.next()).value["price"];
 
+      let previousTime = "1970-01-01T00:00:00Z";
+      let recentTime;
       for await (const value of data) {
         if (value.action === "CREATE") {
           if (value.type === "BALANCE") {
-            currentBalanceOfToken.value = convertWeiToQuantity(
-              value.result["walletValue"],
-              parseInt(decimals),
-            );
+            recentTime = value.result["timestamp"];
+            if (
+              new Date(recentTime).getTime() >= new Date(previousTime).getTime()
+            ) {
+              currentBalanceOfToken.value = convertWeiToQuantity(
+                value.result["walletValue"],
+                parseInt(decimals),
+              );
+              previousTime = recentTime;
+            }
```

Comment:
could we go for something like:
`for await (const {action, type, result} of data) {...`

and 
`cont {timestamp, walletValue) = result` 
etc.

Reply:
done. also did 
const {timestamp, walletValue, price) = result